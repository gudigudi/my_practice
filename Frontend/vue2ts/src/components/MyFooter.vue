<template>
  <div class="footer">
    <p class="info">{{ info }}</p>
    <a-button @click="modify">Modify</a-button>
  </div>
</template>

<script lang="ts">
import { Component, Prop, PropSync, Vue, Emit } from "vue-property-decorator";
import { Button } from "ant-design-vue";

@Component({ components: { "a-button": Button } })
export default class MyFooter extends Vue {
  @Prop(String)
  public readonly msg: string | undefined;

  public info: string = "Click to modify";

  @Emit()
  modify() {
    this.info = "Modified by click";
  }
}
</script>

<style lang="scss">
</style>
