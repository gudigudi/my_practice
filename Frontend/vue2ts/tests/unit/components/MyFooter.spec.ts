import { shallowMount, createWrapper, mount, createLocalVue } from '@vue/test-utils'
import MyFooter from '@/components/MyFooter.vue'
import Button from 'ant-design-vue'

describe('MyFooter.vue', () => {
    it('renders component', () => {
        const wrapper = shallowMount(MyFooter);

        const text = wrapper.find('.info').text();
        const html = wrapper.find('.info').html();
        const classes = wrapper.find('.info').classes();
        const element = wrapper.find('.info').element;

        expect(text).toEqual('Click to modify');
        expect(html).toEqual('<p class="info">Click to modify</p>');
        expect(classes).toEqual(['info']);
        expect(element).toBeInstanceOf(HTMLParagraphElement);

        expect(wrapper).toMatchSnapshot();
    });
    it('should modify the text after calling modify', () => {
        const wrapper = shallowMount(MyFooter);

        (wrapper.vm as any).modify();

        expect(wrapper.vm.$data.info).toEqual('Modified by click');
    });
    it('should modify the text after clicking the button', async () => {
        const wrapper = mount(MyFooter);

        wrapper.find('.ant-btn').trigger('click');
        await wrapper.vm.$nextTick(); // Wait until trigger events have been handled

        const text = wrapper.find('.info').text();
        expect(text).toEqual('Modified by click');
    });
    it('should handle interactions', () => {
        const wrapper = shallowMount(MyFooter, {
            propsData: { info: 'Click to modify' }
        });

        (wrapper.vm as any).modify();
        expect(wrapper.vm.$data.info).toEqual('Modified by click');
        // expect(wrapper.emitted().modify).toEqual([
        //     ['Modified by click']
        // ]);
    });
})
