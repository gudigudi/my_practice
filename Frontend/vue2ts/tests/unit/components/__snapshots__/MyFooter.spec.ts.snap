// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFooter.vue renders component 1`] = `
<div class="footer">
  <p class="info">Click to modify</p>
  <a-button-stub htmltype="button" size="default">Modify</a-button-stub>
</div>
`;
